### PR TITLE
Streamline process of adding JS autocompletes - round 1

### PR DESCRIPTION
**Pull request by TobyRet**
## Context

We have a lot of autocompletes and autosuggests on Apply and this is becoming increasingly difficult to manage. How could this be streamlined?

## Changes proposed in this pull request

- Autosuggests and autocompletes are functionally different. This PR now makes that distinction by name spacing autosuggests and autocompletes and renaming files and variables where appropriate

## Guidance to review

- Make sense?
- Be aware we do not currently have specs for testing autocomplete behaviour (out of scope) so need to rely on manual checking. I've gone through and manually tested the autocompletes and they look ok to me, but worth double checking?

## Link to Trello card

https://trello.com/c/0UYEISRa/2667-streamline-process-of-adding-js-autocompletes-round-1
## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/frontend/packs/application-candidate.js [hunk 1]
```diff
@@ -1,20 +1,21 @@
+import initNationalityAutocomplete from "./autocompletes/nationality-autocomplete";
```
Comment (davidgisbey) [inline app/frontend/packs/application-candidate.js:1]: Is this meant to be up here?

**Comment by davidgisbey** [comment]
Code looks good. I'm just going to manually test they're all working and I'll approve

File: app/frontend/packs/application-candidate.js [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
-import initOtherQualificationsSubjectAutocomplete from "./other-qualifications-subject-autocomplete";
+import initProvidersAutocomplete from "./autocompletes/providers-autocomplete";
+import initCoursesAutocomplete from "./autocompletes/courses-autocomplete";
+import initCountryAutocomplete from "./autocompletes/country-autocomplete";
+import initDegreeGradeAutosuggest from "./autosuggests/degree-grade-autosuggest";
+import initDegreeInstitutionAutosuggest from "./autosuggests/degree-institution-autosuggest";
+import initDegreeSubjectAutosuggest from "./autosuggests/degree-subject-autosuggest";
+import initDegreeTypeAutosuggest from "./autosuggests/degree-type-autosuggest";
+import initGcseGradeAutocomplete from "./autocompletes/gcse-grade-autocomplete";
+import initIeltsBandScoreAutocomplete from "./autocompletes/ielts-band-score-autocomplete";
```
Comment (davidgisbey) [inline app/frontend/packs/application-candidate.js:15]: We don't use this autocomplete anymore I think we remove it.

File: app/frontend/packs/application-candidate.js [hunk 3]
```diff
@@ -1,20 +1,21 @@
...
-import initOtherQualificationsGradeAutocomplete from "./other-qualifications-grade-autocomplete";
-import initOtherQualificationsSubjectAutocomplete from "./other-qualifications-subject-autocomplete";
+import initProvidersAutocomplete from "./autocompletes/providers-autocomplete";
+import initCoursesAutocomplete from "./autocompletes/courses-autocomplete";
+import initCountryAutocomplete from "./autocompletes/country-autocomplete";
+import initDegreeGradeAutosuggest from "./autosuggests/degree-grade-autosuggest";
+import initDegreeInstitutionAutosuggest from "./autosuggests/degree-institution-autosuggest";
+import initDegreeSubjectAutosuggest from "./autosuggests/degree-subject-autosuggest";
+import initDegreeTypeAutosuggest from "./autosuggests/degree-type-autosuggest";
+import initGcseGradeAutocomplete from "./autocompletes/gcse-grade-autocomplete";
```
Comment (davidgisbey) [inline app/frontend/packs/application-candidate.js:14]: this can go to

**Review by davidgisbey — APPROVED** [review]
It's all working nicely. I think a couple of these can be removed as we don't autosuggest for GCSE grade or ielts score anymore.

**Review by stevehook — APPROVED** [review]
Did a quick test of the autocompletes on the review app and all good.

File: app/frontend/packs/application-candidate.js (hunk 1 above)
Comment (TobyRet) [reply app/frontend/packs/application-candidate.js:1]: 👍

File: app/frontend/packs/application-candidate.js (hunk 3 above)
Comment (TobyRet) [reply app/frontend/packs/application-candidate.js]: 👍

**Review by george-bit — APPROVED** [review]

**Pull request merged**